Extensions/BitmapText/bitmaptextruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * @constructor
+ * @memberof gdjs
+ * @class BitmapFontManager
+ */
+gdjs.BitmapFontManager = {
+  _fontUsed: [],
+  _fontTobeUnloaded: [],
+};
+
+/**
+ * When an object use an bitmapFont register the slug of the font,
+ * and how many objects use it.
+ * @param {string} name Slug of the bitmapFont
+ */
+gdjs.BitmapFontManager.setFontUsed = function (name) {
+  if (gdjs.BitmapFontManager._fontUsed[name]) {
+    gdjs.BitmapFontManager._fontUsed[name].objectsUsingTheFont =
+      gdjs.BitmapFontManager._fontUsed[name].objectsUsingTheFont + 1 || 1;
+  } else {
+    gdjs.BitmapFontManager._fontUsed[name] = {
+      objectsUsingTheFont: 1,
+    };
+  }
+};
+
+/**
+ * When an bitmapText object is removed, decrease the count of objects related to the font in the manager.
+ * When an font is unused the font goes in a temporary cache where if there is more than 10 fonts, the lastest is deleted.
+ * @param {string} name Slug of the bitmapFont
+ */
+gdjs.BitmapFontManager.removeFontUsed = function (name) {
+  if (gdjs.BitmapFontManager._fontUsed[name]) {
+    gdjs.BitmapFontManager._fontUsed[name].objectsUsingTheFont =
+      gdjs.BitmapFontManager._fontUsed[name].objectsUsingTheFont - 1 || 0;
+  }
+
+  if (gdjs.BitmapFontManager._fontUsed[name].objectsUsingTheFont <= 0) {
+    gdjs.BitmapFontManager._fontTobeUnloaded[name] =
+      gdjs.BitmapFontManager._fontUsed[name];
+    delete gdjs.BitmapFontManager._fontUsed[name];
+
+    const countFontToBeUnloaded = Object.keys(
+      gdjs.BitmapFontManager._fontTobeUnloaded
+    ).length;
+    const lastFontStored = Object.keys(gdjs.BitmapFontManager._fontTobeUnloaded)[Object.keys(gdjs.BitmapFontManager._fontTobeUnloaded).length - 1];
```

Comment:
The JavaScript objects don't necessarily store the last added key as the last in memory. Instead, consider using an array containing just the font name (an array of strings, the strings being the slugs).

```
// Add the font name at the last position of the cache.
if (!gdjs.BitmapFontManager._fontTobeUnloaded.includes(name))
  gdjs.BitmapFontManager._fontTobeUnloaded.push(name);

if (gdjs.BitmapFontManager._fontTobeUnloaded.length > 10) {
  // Remove the first font (i.e: the oldest one)
  const oldestUnloadedFont = gdjs.BitmapFontManager._fontTobeUnloaded.shift();
  PIXI.BitmapFont.uninstall(oldestUnloadedFont);
}
```